### DISCUSSION TITLE
Palm Springs Hydrojet 54144

### DISCUSSION THREAD

**Discussion by BigDougSmith** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#74] [General]
Hi Folk.

I'm intending to set this up on a Palm Springs hydrojet pump. It's one of the square ones with the 4 wire connection.

Interestingly, I've got two pumps and they're both slightly different.

The one that came with the hot tub is labelled 54144 ...

![image](https://user-images.githubusercontent.com/84990268/120028107-2c383400-bfec-11eb-9dd9-4b4f8c0ae388.png)

and the one that I bought on ebay for spares is also labelled 54144 (don't you just love it when different things are given the same ID)...

![image](https://user-images.githubusercontent.com/84990268/120028252-5984e200-bfec-11eb-9b57-94afd4dfe6c7.png)

Note that the ebay one has an extra button in the middle, any idea what that is supposed to do, I've had a look at the manuals on the bestway website and it doesn't seem to match anything there. Is it a hydrojet "pro"?

They both have the same two hose connector layout, so I guess they can't be all that different!

I think I'll build using stripboard (as it is "to hand") but the PCBs folk have been making look really neat, nice job.

Hopefully the  4 wire code will work perfectly, however I guess I'll do a bit of bench testing. It will be interesting to see if both pumps function in the same way.

I've had a go with PlatformIO, and managed to program the NodeMCU module, and upload the filesystem over USB, this all works great. OTA programming works fine for the software, but Upload Filesystem Image OTA seems to fail at the very last step. Is this just me ?

Anyhow, hopefully I'll have this up and running over the next couple of days. I get cheap power between midnight and 4am, so the idea is to heat the tub when it is really cheap and only need a little booster when it is time go get in. This looks like a perfect solution.

Cheers,

Doug.

**Comment by visualapproach** [comment]
Hi
Never seen that button. The only hydrojet pro I've seen had both bubbles and jets. Looks like it says something like 'water recirculation'. 
I'll see if LittleFS OTA is working here.
Excited to hear about your findings on the pumps.

**Reply by visualapproach** [reply]
It seems "upload filesystem image OTA" is not working, but "upload filesystem image" works OVER THE AIR! Strange. You need to set these params in platformio.ini though:
upload_protocol = espota
upload_port = 192.168.x.y
upload_flags = 
	--auth=esp8266

**Reply by BigDougSmith** [reply]
The manual (Palm-Springs-HJ-2019.pdf) shows exactly what you say, 

![image](https://user-images.githubusercontent.com/84990268/120065792-1e75c380-c06b-11eb-8227-cac463450dfa.png)

I think the ebay pump is from 2018, but it's not got the additional air pipe, so can't blow bubbles. I'll continue to dig around and see if I can find an older manual online somewhere.

Thanks for looking at the OTA issue, I'll have a look and see if "upload filesystem image" uses OTA for me too :)

Cheers,

Doug.

**Reply by gavinchan55** [reply]
@BigDougSmith did you get this working successfully? Got the same model, and I'm crossing my fingers 😆

**Comment by marjakit** [comment]
The extra button is to use salt instead of chlorine tablets to clean it. We had that pump originally and it broke 3 times in the warranty period and eventually Bestways sent us the newer replacement pump that no longer has that button. 

Hope that helps.
